src/sync/sync_session.cpp

```diff
@@ -125,10 +121,12 @@ struct sync_session_states::Active : public SyncSession::State {
         // Register all the pending wait-for-completion blocks. This can
         // potentially add a redundant callback if we're coming from the Dying
         // state, but that's okay (we won't call the user callbacks twice).
-        if (!session.m_upload_completion_callbacks.empty())
-            session.add_completion_callback(_impl::SyncProgressNotifier::NotifierType::upload);
-        if (!session.m_download_completion_callbacks.empty())
-            session.add_completion_callback(_impl::SyncProgressNotifier::NotifierType::download);
+        SyncSession::CompletionCallbacks callbacks_to_register;
+        std::swap(session.m_completion_callbacks, callbacks_to_register);
+
+        for (auto& [ id, callback_tuple ] : callbacks_to_register) {
```

Comment:
Spacing on structured binding is inconsistent here; some have spaces in [] and some don't. I think no spaces is the normal thing?